ublox/src/ubx_packets/packets.rs

```diff
@@ -394,14 +530,574 @@ pub enum MapMatchingStatus {
+                rtcm_corr_used: (long & 0b11111111111111111) != 0,
+                slas_corr_used: (long & 0b111111111111111111) != 0,
+                spartn_corr_used: (long & 0b1111111111111111111) != 0,
+                pr_corr_used: (long & 0b11111111111111111111) != 0,
+                cr_corr_used: (long & 0b111111111111111111111) != 0,
+                do_corr_used: (long & 0b1111111111111111111111) != 0,
+                clas_corr_used: (long & 0b11111111111111111111111) != 0,
+            }
+        }
+    }
+
+    /// Signal quality indicator
+    #[ubx_extend]
+    #[ubx(from, rest_reserved)]
+    #[repr(u8)]
+    #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+    pub enum QualityIndicator {
+        NoSignal = 0,
+        SearchingSignal = 1,
+        SignalAcquired = 2,
+        SignalDetectedButUnusable = 3,
+        CodeLockedAndTimeSynchronized = 4,
+        CodeAndCarrierLockedAndTimeSynchronized = 5,
+        CodeAndCarrierLockedAndTimeSynchronizedAndPseudorangeValid = 6,
+        CodeAndCarrierLockedAndTimeSynchronizedAndPseudorangeAndCodePhaseValid = 7,
+    }
+
+    /// Signal health flag
+    #[ubx_extend]
+    #[ubx(from, rest_reserved)]
+    #[repr(u8)]
+    #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+    pub enum Health {
+        Unknown = 0,
+        Healthy = 1,
+        Unhealthy = 2,
+    }
+
+    /// Orbit source
+    #[ubx_extend]
+    #[ubx(from, rest_reserved)]
+    #[repr(u8)]
+    #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+    pub enum OrbitSource {
+        NoOrbitInformation = 0,
+        EphemerIsUsed = 1,
+        AlmanacIsUsed = 2,
+        AssistNowOfflineOrbitIsUsed = 3,
+        AssistNowAutonomousOrbitIsUsed = 4,
+        OtherOrbitInformationIsUsed = 5,
+        OtherOrbitInformationIsUsed2 = 6,
+        OtherOrbitInformationIsUsed3 = 7,
+    }
+
+    pub(crate) fn convert_to_payload(bytes: &[u8]) -> Vec<SatInfo> {
+        let mut res = Vec::new();
+        let mut iter = bytes.iter();
+        while iter.len() >= 12 {
+            let gnssid = iter.next().unwrap();
+            let svid = iter.next().unwrap();
+            let cno = iter.next().unwrap();
+            let elev = iter.next().unwrap();
+            let azim = [*iter.next().unwrap(), *iter.next().unwrap()];
+            let prres = [*iter.next().unwrap(), *iter.next().unwrap()];
+            let flags = [
+                *iter.next().unwrap(),
+                *iter.next().unwrap(),
+                *iter.next().unwrap(),
+                *iter.next().unwrap(),
+            ];
+            res.push(SatInfo {
+                gnssid: *gnssid,
+                svid: *svid,
+                cno: *cno,
+                elev: *elev as i8,
+                azim: i16::from_le_bytes(azim),
+                prres: i16::from_le_bytes(prres),
+                flags: NavServiceBitFlags::from_u32(u32::from_le_bytes(flags)),
+            });
+        }
+        res
```

Comment:
```suggestion
        bytes
            .chunks_exact(12)
            .map(|bytes| {
                let mut iter = bytes.iter();
                let gnssid = *iter.next().unwrap();
                let svid = *iter.next().unwrap();
                let cno = *iter.next().unwrap();
                let elev = *iter.next().unwrap();
                let azim = [*iter.next().unwrap(), *iter.next().unwrap()];
                let prres = [*iter.next().unwrap(), *iter.next().unwrap()];
                let flags = [
                    *iter.next().unwrap(),
                    *iter.next().unwrap(),
                    *iter.next().unwrap(),
                    *iter.next().unwrap(),
                ];
                SatInfo {
                    gnssid,
                    svid,
                    cno,
                    elev: elev as i8,
                    azim: i16::from_le_bytes(azim),
                    prres: i16::from_le_bytes(prres),
                    flags: NavServiceBitFlags::from_u32(u32::from_le_bytes(flags)),
                }
            })
            .collect()
```

This is able to preallocate the `Vec`